internal/internal_event_handlers_test.go

```diff
@@ -402,3 +406,322 @@ func TestProcessQuery_KnownQueryTypes(t *testing.T) {
+					callback: func(result []byte, err error) {
+						var customErr *CustomError
+						assert.ErrorAs(t, err, &customErr)
+						assert.Equal(t, customErr.reason, "test-cause")
+					},
+				})
+				impl.decisionsHelper.getDecisions(true)
+				impl.decisionsHelper.handleTimerStarted("test-timer")
+				decision.cancel()
+				decision.handleDecisionSent()
+				impl.completeHandler = func(result []byte, err error) {
+					assert.NoError(t, err)
+				}
+			},
+		},
+		{
+			event: &s.HistoryEvent{
+				EventType: s.EventTypeWorkflowExecutionCancelRequested.Ptr(),
+				EventId:   common.Int64Ptr(7),
+				WorkflowExecutionCancelRequestedEventAttributes: &s.WorkflowExecutionCancelRequestedEventAttributes{
+					Cause: common.StringPtr("test-cause"),
+				},
+			},
+			prepareHandler: func(t *testing.T, impl *workflowExecutionEventHandlerImpl) {
+				cancelCalled := false
+				t.Cleanup(func() {
+					if !cancelCalled {
+						t.Error("cancelWorkflow not called")
+						t.FailNow()
+					}
+				})
+				impl.cancelHandler = func() {
+					cancelCalled = true
+				}
+			},
+		},
+		{
+			event: &s.HistoryEvent{
+				EventType: s.EventTypeRequestCancelExternalWorkflowExecutionInitiated.Ptr(),
+				EventId:   common.Int64Ptr(8),
+				RequestCancelExternalWorkflowExecutionInitiatedEventAttributes: &s.RequestCancelExternalWorkflowExecutionInitiatedEventAttributes{
+					DecisionTaskCompletedEventId: common.Int64Ptr(7),
+					Domain:                       common.StringPtr(testDomain),
+					WorkflowExecution: &s.WorkflowExecution{
+						WorkflowId: common.StringPtr("wid"),
+					},
+					ChildWorkflowOnly: common.BoolPtr(true),
+				},
+			},
+			prepareHandler: func(t *testing.T, impl *workflowExecutionEventHandlerImpl) {
+				decision := impl.decisionsHelper.startChildWorkflowExecution(&s.StartChildWorkflowExecutionDecisionAttributes{
+					Domain:     common.StringPtr(testDomain),
+					WorkflowId: common.StringPtr("wid"),
+				})
+				decision.setData(&scheduledChildWorkflow{
+					handled: true,
+				})
+				impl.decisionsHelper.getDecisions(true)
+				impl.decisionsHelper.handleStartChildWorkflowExecutionInitiated("wid")
+				impl.decisionsHelper.handleChildWorkflowExecutionStarted("wid")
+				decision = impl.decisionsHelper.requestCancelExternalWorkflowExecution(testDomain, "wid", "", "", true)
+				decision.setData(&scheduledCancellation{
+					callback: func(result []byte, err error) {
+						var canceledErr *CanceledError
+						assert.ErrorAs(t, err, &canceledErr)
+					},
+				})
+				impl.decisionsHelper.getDecisions(true)
+				impl.decisionsHelper.handleRequestCancelExternalWorkflowExecutionInitiated(7, "wid", "")
+				decision.handleDecisionSent()
+			},
+		},
+	} {
+		t.Run(tc.event.EventType.String(), func(t *testing.T) {
+			weh := testWorkflowExecutionEventHandler(t, newRegistry())
+			if tc.prepareHandler != nil {
+				tc.prepareHandler(t, weh)
+			}
+
+			err := weh.ProcessEvent(tc.event, false, false)
+			assert.NoError(t, err)
```

Comment:
Same as above, can we test something more?

Reply:
From the existing method - no.
The assertion happens inside the prepareHandler setup.

Reply:
Added a comment and moved assertion explicitily here.